### ISSUE TITLE
Sometimes ChatClientListenerForwarder returns `CHANNELS_COMPLETED` after `COMPLETED`

### ISSUE THREAD

**Issue by rusmichal** [twilio/twilio-chat-demo-android#95]
### Description

We've noticed that sometime Twilio `ChatClientListenerForwarder` callback brings `CHANNELS_COMPLETED`after the SynchronizationStatus::COMPLETED will show up. it is really hard to reproduce I couldn't have found any patters. The problem is that we are waiting COMPLETED event and we have to add tricky part in order to handle that unexpected behaviour. Please find two file with logs, first one where every things works fine, second file please find logs where the `CHANNELS_COMPLETED` is after `COMPLETED`

### Steps to Reproduce

It happens randomly

#### Expected Behavior

COMPLETED after CHANNELS_COMPLETED

#### Actual Behavior

CHANNELS_COMPLETED after COMPLETED

#### Reproduces how Often

Hard to say, just many times I have to initialize chat client with the same user

#### Logs

[wrong_behaviour.txt](https://github.com/twilio/twilio-chat-demo-android/files/3257688/wrong_behaviour.txt)
[correct_behaviour.txt](https://github.com/twilio/twilio-chat-demo-android/files/3257744/correct_behaviour.txt)

If you need more logs let me know. I can reproduce it but it taks awhile.

#### Chat Android SDK

Twilio Chat SDK version 4.0.4

#### Android API

28
#### Android Device

BRAND OnePlus
DEVICE OnePlus6T
HARDWARE qcom

**Comment by berkus** [comment]
Hi, thanks for the report!

I've heard about this before, the workaround we recommended before is to ignore CHANNELS_COMPLETED arriving after COMPLETED status has been received. But I've filed an internal ticket to fix it properly, will keep you posted.

**Comment by rusmichal** [comment]
Hey, another thoughts.

I'm calling 
`ChatClient.create(cilabsApplication, accessToken, props, InitClientListener(emitter))`
On `InitClientListener` i've got callback where I can take reference to `chatClient`

After that I invoke `this.chatClient?.setListener(this.chatClientListener)` immediately in order to get any callback from chatClient (e.g `Client synchronization update - COMPLETED`)

The interesting thing is that SDK emits two states from `onClientSynchronization`:
```
2019-06-06 10:58:17.280 5062-5200/? I/TWC: [2019-06-06 10:58:17.156] [rtd] [INFO]       482755024112 |        | Router | Message with type 'twilio.sync.event, from channel: twilsock' is being routed to 1 targets
2019-06-06 10:58:17.575 5062-6396/? D/ChatClientListenerForwarder: Client synchronization update - CHANNELS_COMPLETED
2019-06-06 10:58:17.575 5062-6394/? D/ChatClientListenerForwarder: Client synchronization update - COMPLETED
2019-06-06 10:58:17.681 5062-5200/? D/TWC: [2019-06-06 10:58:17.558] [rtd] [DEBUG]       482759185648 | ChatClientListener(native) | onChannelUpdated - channel sync
```
But the method `onClientSynchronization` from `TwilioClientListener` is invoked once. :O

I thought that I've registered that listener too later but I did it in callback from initialization of chatClient.

See my code from `TwilioClientListener`

```
override fun onClientSynchronization(syncStatus: ChatClient.SynchronizationStatus) {
        Log.d("ChatLogger","Client synchronization stats: $syncStatus")
        clientStatusRelay.accept(ChatClientStatus(syncStatus))
    }
```

More logs here:

[emits_once_log.txt](https://github.com/twilio/twilio-chat-demo-android/files/3260882/emits_once_log.txt)

**Comment by berkus** [comment]
Thanks! This is helpful!

We have something to test internally, when it passes tests I will update here.

**Comment by rusmichal** [comment]
hey, any updates?

**Comment by JT9193** [comment]
We are also running into this issue, if you need any additional information or logs, just let me know and I can provide them.

**Comment by Pavel-Cherentsov** [comment]
Sorry for the late response..
The issue is reproduced, so thanks, no need in additional logs.

We've just released Chat SDK with a lot of stability fixes,
and this issue we plan to fix in the following release.

**Comment by rusmichal** [comment]
Thank you @Pavel-Cherentsov it seems that the problem has gone in version 4.2 SDK. Could you confirm that?

**Comment by Pavel-Cherentsov** [comment]
Hi @rusmichal,
We've already fixed this problem, but current 4.2.0 does not yet contain our fix, sorry.
The fix should appear in the next release.

**Comment by Pavel-Cherentsov** [comment]
Should be fixed starting from 4.2.1.
